pkg/utils/bpf-iter-ns/bpf-iter-ns.go

```diff
@@ -55,9 +57,28 @@ func ReadOnCurrentPidNs(iter *link.Iter) ([]byte, error) {
 		return nil, fmt.Errorf("open BPF iterator: %w", err)
 	}
 	defer file.Close()
-	buf, err := io.ReadAll(file)
-	if err != nil {
-		return nil, fmt.Errorf("read BPF iterator: %w", err)
+
+	var b bytes.Buffer
+	r := bufio.NewReader(file)
+	buf := make([]byte, 8192)
+	// we cannot use io.ReadAll() since reading from the BPF iterator file could return more than
+	// one million records are iterated on and the OS will return EAGAIN.
+	// We need to continue reading with another sys call and accumulating data.
+	// Rootcause: https://github.com/torvalds/linux/commit/e679654a704e5bd676ea6446fa7b764cbabf168a
+	for {
+		n, err := r.Read(buf)
+		if n > 0 {
+			b.Write(buf[:n])
+		}
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if errors.Is(err, syscall.EAGAIN) {
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("read BPF iterator: %w", err)
+		}
 	}
 	return buf, err
```

Comment:
This returns the wrong buffer and the wrong error:

```suggestion
	return b.Bytes(), nil
```

This was found by the CI:

```
    bpfstats_test.go:153: 
        	Error Trace:	/home/runner/work/inspektor-gadget/inspektor-gadget/gadgets/bpfstats/test/unit/bpfstats_test.go:153
        	Error:      	[]tests.ExpectedBpfstatsEvent{} does not contain tests.ExpectedBpfstatsEvent{GadgetID:"", GadgetImage:"ttl.sh/80fff3e9a5efc3700748808ebf0e00e8b36f2a53/trace_open:4944-merge", GadgetName:"", MapCount:1, MapMemory:"foo", ProgID:0, ProgName:"", ProgType:"", Runcount:1, Runtime:"foo", Comms:"", Pids:""}
        	Test:       	TestBpfstatsGadget/by_gadget
--- FAIL: TestBpfstatsGadget (0.00s)
    --- FAIL: TestBpfstatsGadget/by_programs (7.89s)
    --- FAIL: TestBpfstatsGadget/by_gadget (7.67s)
```